### DISCUSSION TITLE
How to get gpu id corresponding to each process in 'ddp'?

### DISCUSSION THREAD

**Discussion by xiadingZ** [Lightning-AI/pytorch-lightning#264] [DDP / multi-GPU / multi-node] [answered]
In ddp mode, if I use 8 gpus, then it will creates 8 processes for each gpu, if I want to create new tensor at runtime, can I use `.cuda(self.trainer.root_gpu)`? will it use correct gpu corresponding to its process?

**Comment by williamFalcon** [comment, marked as answer]
yes
